src/System.Private.ServiceModel/src/System/ServiceModel/Security/EncryptedKey.cs

```diff
@@ -0,0 +1,132 @@
+        internal static readonly XmlDictionaryString s_RecipientAttribute = System.IdentityModel.XD.XmlEncryptionDictionary.Recipient;
+
+        private string _carriedKeyName;
+        private string _recipient;
+        private ReferenceList _referenceList;
+        private byte[] _wrappedKey;
+
+        public string CarriedKeyName
+        {
+            get { return _carriedKeyName; }
+            set { _carriedKeyName = value; }
+        }
+
+        public string Recipient
+        {
+            get { return _recipient; }
+            set { _recipient = value; }
+        }
+
+        public ReferenceList ReferenceList
+        {
+            get { return _referenceList; }
+            set { _referenceList = value; }
+        }
+
+        protected override XmlDictionaryString OpeningElementName
+        {
+            get { return s_ElementName; }
+        }
+
+        protected override void ForceEncryption()
+        {
+            // no work to be done here since, unlike bulk encryption, key wrapping is done eagerly
+        }
+
+        public byte[] GetWrappedKey()
+        {
+            if (this.State == EncryptionState.New)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.BadEncryptionState)));
+            }
+            return _wrappedKey;
+        }
+
+        public void SetUpKeyWrap(byte[] wrappedKey)
+        {
+            if (this.State != EncryptionState.New)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.BadEncryptionState)));
+            }
+            if (wrappedKey == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("wrappedKey");
+            }
+            _wrappedKey = wrappedKey;
+            this.State = EncryptionState.Encrypted;
+        }
+
+        protected override void ReadAdditionalAttributes(XmlDictionaryReader reader)
+        {
+            _recipient = reader.GetAttribute(s_RecipientAttribute, null);
+        }
+
+        protected override void ReadAdditionalElements(XmlDictionaryReader reader)
+        {
+            throw ExceptionHelper.PlatformNotSupported();   // Issue #31 in progress
+
+            //if (reader.IsStartElement(ReferenceList.ElementName, EncryptedType.NamespaceUri))
+            //{
+            //    this.referenceList = new ReferenceList();
+            //    this.referenceList.ReadFrom(reader);
+            //}
+            //if (reader.IsStartElement(CarriedKeyElementName, EncryptedType.NamespaceUri))
+            //{
+            //    reader.ReadStartElement(CarriedKeyElementName, EncryptedType.NamespaceUri);
+            //    this.carriedKeyName = reader.ReadString();
+            //    reader.ReadEndElement();
+            //}
```

Comment:
I'll can enable all of it except the call to ReadString which is in Net Standard 2.0.  I'm going to add "NET Standard 2.0" to all comments where we disable something pending taking a dependency on the new API's coming into CoreFx's NET Standard 2.0.  I may be able to use the dev/api feed once we create our own dev/api branch, but I'm hanging in master just to stay in sync with its changes.

---
In reply to: [78486674](https://github.com/dotnet/wcf/pull/1515#discussion_r78486674) [](ancestors = 78486674)